pkg/agent/controller/networkpolicy/fqdn.go

```diff
@@ -594,38 +637,54 @@ func (f *fqdnController) runRuleSyncTracker(stopCh <-chan struct{}) {
 }
 
 // parseDNSResponse returns the FQDN, IP query result and lowest applicable TTL of a DNS response.
-func (f *fqdnController) parseDNSResponse(msg *dns.Msg) (string, map[string]net.IP, uint32, error) {
+func (f *fqdnController) parseDNSResponse(msg *dns.Msg) (string, map[string]ipWithTTL, error) {
 	if len(msg.Question) == 0 {
-		return "", nil, 0, fmt.Errorf("invalid DNS message")
+		return "", nil, fmt.Errorf("invalid DNS message")
 	}
 	fqdn := strings.ToLower(msg.Question[0].Name)
-	lowestTTL := uint32(math.MaxUint32) // a TTL must exist in the RRs
-	responseIPs := map[string]net.IP{}
+	// Case 1: In the upcoming patch an admin would set this to a  maximum value, which will be read from the configuration.
+	maxConfiguredTTL := uint32(math.MaxUint32) // a TTL must exist in the RRs
+	responseIPs := map[string]ipWithTTL{}
+	currentTime := time.Now()
 	for _, ans := range msg.Answer {
 		switch r := ans.(type) {
 		case *dns.A:
 			if f.ipv4Enabled {
-				responseIPs[r.A.String()] = r.A
-				if r.Header().Ttl < lowestTTL {
-					lowestTTL = r.Header().Ttl
+				// So, using case 1 above , we may not make below comparison and just assign the max value as expirationTime ; ignoring the incoming dns ttl.
+				var expirationTime uint32
+				if r.Header().Ttl < maxConfiguredTTL {
```

Comment:
`expirationTime = minTime(maxConfiguredTTL, r.Header().Ttl)` if you just make the two method above consumable outside the function?

Reply:
@Dyanngg  That seems like a nice improvement, but the ```minTime()``` function  (now named as ```earlierOf()``` as suggested ), 

```go
	minTime := func(t1, t2 time.Time) time.Time {
		if t1.Before(t2) {
			return t1
		}
		return t2
	}
```

 actually accepts and returns values  of type ```time.Time``` as per the requirement of usage in ```onDNSResponse()``` function. I mean at this point i have kept them outside the ```onDNSResponse()``` function as they might be useful in future, but should i keep this quoted part of code as is or try enclosing it into a seperate function like this 

```go
func minTTLtime(ttl1, ttl2 uint32) uint32 {
    if ttl1 < ttl2 {
        return ttl1
    }
    return ttl2
}
```